bin/ons_csv_to_ctb_json_load.py

```diff
@@ -283,24 +299,34 @@ def datasets(self):
             # If the dataset is public then ensure that there is at least one classification and
             # that all the classifications are also public.
             if dataset['Security_Mnemonic'] == PUBLIC_SECURITY_MNEMONIC:
+                drop_dataset = False
                 if not dataset_variables.classifications:
-                    raise ValueError(
+                    self.recoverable_error(
                         f'Reading {self.full_filename(filename)}:{row_num} {dataset_mnemonic} '
                         'has no associated classifications or geographic variable')
+                    drop_dataset = True
```

Comment:
Is it worth mentioning in error that dataset will be dropped?

Reply:
This is reported in a separate message on line 326. Wanted this in a separate message because it doesn't make sense to say it's being dropped if we exit early.